iceoryx2/tests/service_request_response_tests.rs

```diff
@@ -376,6 +376,89 @@ mod service_request_response {
         }
     }
 
+    #[test]
+    fn sent_requests_from_disconnected_clients_can_be_received<Sut: Service>() {
+        let test_args = Args {
+            number_of_active_requests: 8,
+            enable_fire_and_forget: true,
+            ..Default::default()
+        };
+
+        let mut test = TestFixture::<Sut>::new(test_args);
+        assert_that!(test.servers[0].receive(), is_ok);
+
+        for n in 0..test_args.number_of_active_requests {
+            test.clients[0].send_copy(n).unwrap();
+        }
+
+        // disconnect all clients
+        test.clients.clear();
+
+        for n in 0..test_args.number_of_active_requests {
+            let active_request = test.servers[0].receive().unwrap().unwrap();
+            assert_that!(*active_request.payload(), eq n);
+            assert_that!(active_request.is_connected(), eq false);
+        }
+    }
+
+    #[test]
+    fn sent_requests_from_disconnected_clients_are_not_received_without_fire_and_forget<
+        Sut: Service,
+    >() {
+        let test_args = Args {
+            number_of_active_requests: 8,
+            enable_fire_and_forget: false,
+            ..Default::default()
+        };
+
+        let mut test = TestFixture::<Sut>::new(test_args);
+        assert_that!(test.servers[0].receive(), is_ok);
+
+        for n in 0..test_args.number_of_active_requests {
+            test.clients[0].send_copy(n).unwrap();
+        }
+
+        // disconnect all clients
+        test.clients.clear();
+
+        let active_request = test.servers[0].receive().unwrap();
+        assert_that!(active_request, is_none);
+    }
+
+    #[test]
+    fn sent_requests_from_disconnected_clients_are_received_first<Sut: Service>() {
```

Comment:
I guess it is a deliberate decision to have the requests from the disconnected clients first?

There is probably anyway no way to detect the order of the requests from different clients

Reply:
No not yet. Maybe we introduce priorities at some point in time ... then this will become interesting again.